net/net.go

```diff
@@ -866,41 +800,54 @@ func (n *net) ConnectApp(a app.App, id thread.ID) (*app.Connector, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error getting thread %s: %v", id, err)
 	}
-	con, err := app.NewConnector(a, n, info, func(ctx context.Context, id thread.ID) (<-chan core.ThreadRecord, error) {
-		return n.subscribe(ctx, map[thread.ID]struct{}{id: {}})
-	})
+	con, err := app.NewConnector(a, n, info)
 	if err != nil {
 		return nil, fmt.Errorf("error making connector %s: %v", id, err)
 	}
-	n.apps[id] = con.Token()
+	n.connectors[id] = con
 	return con, nil
 }
 
-// validateAPIToken checks if the thread is backing an app,
-// and if so, returns whether or not the token matches the app
-// connector's token.
-func (n *net) validateAPIToken(id thread.ID, token core.Token) bool {
-	t, ok := n.apps[id]
+// @todo: Handle thread ACL checks against ID and readOnly.
+func (n *net) Validate(id thread.ID, token thread.Token, readOnly bool) (thread.PubKey, error) {
+	if err := id.Validate(); err != nil {
+		return nil, err
+	}
+	return token.Validate(n.getPrivKey())
+}
+
+// getConnector returns the connector tied to the thread if it exists
+// and whether or not the token is valid.
+func (n *net) getConnector(id thread.ID, token core.Token) (*app.Connector, bool) {
```

Comment:
Connectors proof ownership of a thread by passing a unique `core.Token` (random bytes).